dashboard/app/models/user.rb

```diff
@@ -219,7 +219,7 @@ class User < ActiveRecord::Base
   has_many :districts, through: :districts_users
 
   has_many :authentication_options, dependent: :destroy
-  belongs_to :primary_authentication_option, class_name: 'AuthenticationOption'
+  belongs_to :primary_contact_info, class_name: 'AuthenticationOption'
```

Comment:
I just noticed we have `belongs_to` on both sides of this 1:1 relationship.  The [ActiveRecord Association guide](http://guides.rubyonrails.org/association_basics.html#choosing-between-belongs-to-and-has-one) suggests you should have `has_one` on one end and `belongs_to` on the other, and this affects the creation of foreign keys.

I'm curious if this was intentional, and what folks think about it.  Changing it may add time to this migration, which might be a good reason not to.

Reply:
interesting, i hadn't noticed that. it seems like the foreign key should go on `AuthenticationOption` so that the logic would be `AuthenticationOption belongs_to :user` and `User has_one :primary_contact_info`. either way, i think changing the relationship and dropping the extra foreign key could be done in a separate PR just to isolate changes

Reply:
It seems like the way it *should be* is that a user `has_one :primary_contact_info` and a primary_contact_info `belongs_to :user`; however, the problem there would be that `has_one` assumes that there is *only one* `AuthenticationOption` that belongs to user. That's not really the case here; a user actually `has_many :authentication_options`, so we really need to store the primary ID in the table. `belongs_to` is just used to make the lookup work.

I have no idea if this is the right way to mark a particular record as "special" in a `has_many` relationship, it seems semantically wrong, but does work technically.

Reply:
Fascinating!  I understand now.  Thanks!